client/src/components/SignUp.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import axios from 'axios';
+
+import { useInput } from '../utilities/useInput';
+
+const SignUp = ({ history }) => {
+  const usernameSignUp = useInput();
+  const passwordSignUp = useInput();
+  const departmentSignUp = useInput();
+
+  const signUp = async e => {
+    e.preventDefault();
+    try {
+      const response = await axios.post('auth/register', {
+        username: usernameSignUp.value,
+        password: passwordSignUp.value,
+        department: departmentSignUp.value
+      });
+      console.log(response.data.message);
+      usernameSignUp.setValue('');
+      passwordSignUp.setValue('');
+      departmentSignUp.setValue('');
+      history.push('/users');
```

Comment:
you push to view the users upon a successful, but it looks like you still need to send a token back from your register endpoint in your API, and then set it into local storage in this function as well -- otherwise a new user just gets sent to the users page, where they are then told they need to login and have to click the login navlink themselves to get there, tsk tsk